apps/nowcasting-app/components/charts/remix-line.tsx

```diff
@@ -0,0 +1,197 @@
+        strokeWidth="1"
+        strokeDasharray="3 3"
+        fill="none"
+        fillOpacity="1"
+        x1={x}
+        y1={yy - 50}
+        x2={x}
+        y2={yy}
+      ></line>
+      <rect
+        x={x - 28}
+        y={yy}
+        width="58"
+        height="30"
+        offset={offset}
+        fill="white"
+      ></rect>
+      <text x={x + 1} y={yy + 21} id="time-now" textAnchor="middle">
+        {value}
+      </text>
+    </g>
+  );
+};
+const RemixLine: React.FC<RemixLineProps> = ({
+  timeOfInterest,
+  data,
+  setTimeOfInterest,
+}) => {
+  const preppedData = data.sort((a, b) =>
+    a.datetimeUtc.localeCompare(b.datetimeUtc)
+  );
+  /** Ensures that the legend is ordered in the same way as the stacked items */
+  function prettyPrintYNumberWithCommas(x) {
+    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+  }
+  function prettyPrintXdate(x) {
+    return x.slice(11, 16);
+  }
+
+  return (
+    <ResponsiveContainer width="100%" height="100%">
+      <ComposedChart
+        width={500}
+        height={400}
+        data={preppedData}
+        margin={{
+          top: 20,
+          right: 20,
+          bottom: 20,
+          left: 20,
+        }}
+        onClick={(e?: { activeLabel?: string }) =>
+          e?.activeLabel && setTimeOfInterest(e.activeLabel)
+        }
+      >
+        <CartesianGrid
+          verticalFill={["#545454", "#6C6C6C"]}
+          fillOpacity={0.5}
+        />
+        <XAxis
+          dataKey="datetimeUtc"
+          tickFormatter={prettyPrintXdate}
+          scale="band"
+          tick={{ fill: "white" }}
+          tickLine={true}
+          interval={11}
+        />
+        <YAxis
+          tickFormatter={(val, i) => prettyPrintYNumberWithCommas(val)}
+          tick={{ fill: "white" }}
+          tickLine={false}
+        />
+
+        <ReferenceLine
+          x={timeOfInterest}
+          stroke="white"
+          strokeWidth={1}
+          strokeDasharray="3 3"
+          label={
+            <CustomizedLabel
+              value={prettyPrintXdate(timeOfInterest)}
+            ></CustomizedLabel>
+          }
+        />
+
+        <Line
+          type="monotone"
+          dataKey="PAST_FORECAST"
+          dot={false}
+          stroke="#FFC425" //yellow
+          strokeWidth={4}
+        />
+        <Line
+          type="monotone"
+          dataKey="FORECAST"
+          dot={false}
+          strokeDasharray="7 7"
+          stroke="#FFC425" //yellow
+          strokeWidth={3}
+        />
+        <Line
+          type="monotone"
+          dataKey="GENERATION"
+          strokeWidth={0}
+          dot={{ fill: "black" }}
+        />
+        <Line
+          type="monotone"
+          dataKey="GENERATION_UPDATED"
+          stroke="black"
+          strokeWidth={3}
+          dot={false}
+        />
+        <Tooltip
+          content={({ payload, label }) => {
+            const data = payload[0]?.payload;
+            if (!data) return <div></div>;
```

Comment:
can it return null, to render nothing, instead of empty div?

Reply:
this would give an error because it's expecting a reactNode (ps : the typescript config is not catching the undefined errors , I'm working on a pr to fix that)